### ISSUE TITLE
Invalid file handle when running in the Mendix Cloud, File is closed twice after downloading.

### ISSUE THREAD

**Issue by Narev** [ArjenLammers/sftp-connector#5]
Since we upgraded our model to Mx8 we are having troubles with the SFTP module. We updated it to the latest version, version 2.0.0 and since then we are encountering problems with our SFTP connetor while getting files. I Added some more logging to the java files to debug and it appears that for some reason the close action is called twice in the Mendix cloud. The weird thing is that when running the application locally, the same model version with the same credentials, everything works just fine.

We are using Mendix version 8.6.3 and the SFTP module 2.0.0. For the test cases i used the default admin overview to simply explore the SFTP server and then download the first file i see in the root.

[Errorlog.log](https://github.com/ArjenLammers/sftp-connector/files/4670198/Errorlog.log)

**Comment by Narev** [comment]
After some scourging of the web i found this [example](https://www.programcreek.com/java-api-examples/?api=net.schmizz.sshj.sftp.RemoteFile) (example 2) that explains how to use the ReadAheadRemoteFileInputStream. It contains the following snippet which i think is exactly what this issue is about, perhaps it would be good to insert this as well to prevent a double close and thus an sftp exception.
```
private final AtomicBoolean close = new AtomicBoolean();

            @Override
            public void close() throws IOException {
                if(close.get()) {
                    log.warn(String.format("Skip double close of stream %s", this));
                    return;
```

**Comment by ArjenLammers** [comment]
Hi Narev, thanks for submitting this.
However, I've solved the issue in a different manner.
It will be fixed in release 2.1.

Kind regards,
Arjen
